drivers/i2c/i2c_mchp_xec.c

```diff
@@ -519,13 +527,17 @@ static void i2c_xec_bus_isr(void *arg)
 		if (MCHP_I2C_SMB_STS_RO(ba) & MCHP_I2C_SMB_STS_LRB_AD0) {
 			MCHP_I2C_SMB_DATA(ba) = dummy;
 		} else {
-			slave_cb->read_processed(data->slave_cfg, &val);
+			if (slave_cb->read_processed) {
+				slave_cb->read_processed(data->slave_cfg, &val);
+			}
 			MCHP_I2C_SMB_DATA(ba) = val;
 		}
 	} else {
 		val = MCHP_I2C_SMB_DATA(ba);
 		/* TODO NACK Master */
```

Comment:
@rgundi Don't you plan to address the TODO comment in this PR?  Everything else looks good

Reply:
Well.. An I2C slave should send a NACK to the master only if it is busy doing something for a long time (where mere clock stretching is not sufficient). It is definitely not needed for the application that i am pursuing right now. So, i don't plan to address it at least in this PR.